tap_github/client.py

```diff
@@ -85,55 +81,65 @@ def get_url_params(
                 params["since"] = since
         return params
 
-    def _request_with_backoff(
-        self, prepared_request, context: Optional[dict]
-    ) -> requests.Response:
-        """Override private method _request_with_backoff to account for expected 404 Not Found erros."""
-        # TODO - Adapt Singer
-        response = self.requests_session.send(prepared_request)
-        if self._LOG_REQUEST_METRICS:
-            extra_tags = {}
-            if self._LOG_REQUEST_METRIC_URLS:
-                extra_tags["url"] = cast(str, prepared_request.path_url)
-            self._write_request_duration_log(
-                endpoint=self.path,
-                response=response,
-                context=context,
-                extra_tags=extra_tags,
-            )
+    def validate_response(self, response: requests.Response) -> None:
+        """Validate HTTP response.
+
+        In case an error is tolerated, continue without raising it.
+
+        In case an error is deemed transient and can be safely retried, then this
+        method should raise an :class:`singer_sdk.exceptions.RetriableAPIError`.
+
+        Args:
+            response: A `requests.Response`_ object.
+
+        Raises:
+            FatalAPIError: If the request is not retriable.
+            RetriableAPIError: If the request is retriable.
+
+        .. _requests.Response:
+            https://docs.python-requests.org/en/latest/api/#requests.Response
+        """
         if response.status_code in self.tolerated_http_errors:
             self.logger.info(
-                "Request returned a tolerated error for {}".format(prepared_request.url)
+                "Request returned a tolerated error for {}".format(response.url)
```

Comment:
splitting hairs, but I'd rephrase that as `...tolerated status code...` just so that we don't have a million "errors" when grepping in logs :)